src/libs/actions/Task.js

```diff
@@ -125,10 +125,96 @@ function createTaskAndNavigate(currentUserEmail, parentReportID, title, descript
     Navigation.navigate(ROUTES.getReportRoute(optimisticTaskReport.reportID));
 }
 
+/**
+ * Sets the task editing property in the store to true
+ *
+ * @params {Object} report
+ */
+
+function beginEditingTask() {
+    Onyx.merge(ONYXKEYS.TASK, {isEditing: true});
+}
+
+/** Sets the task editing property in the store to false */
+function endEditingTask() {
+    Onyx.merge(ONYXKEYS.TASK, {isEditing: false});
+}
+
+// import ReportUtils from '../ReportUtils';
+
+/**
+ * @function editTask
+ * @param {object} report
+ * @param {string} ownerEmail
+ * @param {string} title
+ * @param {string} description
+ * @param {string} assignee
+ * @returns {object} action
+ *
+ */
+
+function editTaskAndNavigate(report, ownerEmail, title, description, assignee) {
+    console.log('parameters', report, ownerEmail, title, description, assignee);
+
+    const editTaskReportAction = ReportUtils.buildOptimisticEditedTaskReportAction(ownerEmail);
+
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.reportID}`,
+            value: {[editTaskReportAction.reportActionID]: editTaskReportAction},
+        },
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
+            value: {
+                ...report,
+                reportName: title || report.reportName,
+                description: description || report.description,
+                assignee: assignee || report.assignee,
+            },
+        },
+    ];
+    const successData = [];
+    const failureData = [
+        // {
+        //     onyxMethod: Onyx.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.reportID}`,
+        //     value: {[editTaskReportAction.reportActionID]: {pendingAction: null}},
+        // },
+        // {
+        //     onyxMethod: Onyx.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
+        //     value: {reportName: report.reportName, description: report.description, assignee: report.assignee},
+        // },
+    ];
+
+    console.log(optimisticData);
+
+    API.write('EditTask', {
+        taskID: report.reportID,
+        title,
+        description,
+        assignee,
+    }, [optimisticData, successData, failureData]);
+
```

Comment:
FYI: I believe what we're deciding to do is always passing the parameters regardless of what is updated

So always passing in the latest or updated title, description, assignee